### DISCUSSION TITLE
error on LAMMPS with DPLR

### DISCUSSION THREAD

**Discussion by ryujh28** [deepmodeling/deepmd-kit#1927] [Q&A]
Hi,

All my DeepMD molecular dynamics simulations were done well with LAMMPS.
This time, I'm trying to run a molecular dynamics simulation with LAMMPS with DPLR potential, and I'm facing an error code given below.

I already tried to fix it by recompiling the DeepMD and LAMMPS.
The dw model and energy model were retrained as well. 

I used DeePMD-kit v2.1.1

Do you have any idea to fix this error?
Thanks for reading

Error code is presented below

Your simulation uses code contributions which should be cited:
- USER-DEEPMD package:
The log file lists these citations in BibTeX format.

CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE

PPPM initialization ...
  using 12-bit tables for long-range coulomb (../kspace.cpp:342)
  G vector (1/distance) = 0.4
  grid = 20 20 20
  stencil order = 5
  estimated absolute RMS force accuracy = 0.059750338
  estimated relative force accuracy = 0.0041494313
  using double precision MKL FFT
  3d grid and FFT values/proc = 24389 8000
WARNING: Interaction between types 1 and 3 is set with deepmd, but will be ignored.
 Deepmd model has only 2 types, it only computes the mulitbody interaction of types: 1-2. (../pair_deepmd.cpp:1077)
WARNING: Interaction between types 2 and 3 is set with deepmd, but will be ignored.
 Deepmd model has only 2 types, it only computes the mulitbody interaction of types: 1-2. (../pair_deepmd.cpp:1077)
WARNING: Interaction between types 3 and 3 is set with deepmd, but will be ignored.
 Deepmd model has only 2 types, it only computes the mulitbody interaction of types: 1-2. (../pair_deepmd.cpp:1077)
Generated 0 of 3 mixed pair_coeff terms from geometric mixing rule
Neighbor list info ...
  update every 1 steps, delay 0 steps, check yes
  max neighbors/atom: 2000, page size: 100000
  master list distance cutoff = 8
  ghost atom cutoff = 8
  binsize = 4, bins = 3 3 3
  1 neighbor lists, perpetual/occasional/extra = 1 0 0
  (1) pair deepmd, perpetual
      attributes: full, newton on
      pair build: full/bin
      stencil: full/bin/3d
      bin: standard
Setting up Verlet run ...
  Unit style    : metal
  Current step  : 0
  Time step     : 0.001
INTERNAL: 2 root error(s) found.
  (0) INTERNAL: Operation received an exception: DeePMD-kit Error: CUDA Assert, in file /home/tools/LAMMPS-DeePMD-OCTP/deepmd-kit-v2.1.4/source/op/custom_op.cc:17
         [[{{node ProdEnvMatA}}]]
         [[o_force/_29]]
  (1) INTERNAL: Operation received an exception: DeePMD-kit Error: CUDA Assert, in file /home/tools/LAMMPS-DeePMD-OCTP/deepmd-kit-v2.1.4/source/op/custom_op.cc:17
         [[{{node ProdEnvMatA}}]]
0 successful operations.
0 derived errors ignored.
ERROR: DeePMD-kit Error: TensorFlow Error: INTERNAL: 2 root error(s) found.
  (0) INTERNAL: Operation received an exception: DeePMD-kit Error: CUDA Assert, in file /home/tools/LAMMPS-DeePMD-OCTP/deepmd-kit-v2.1.4/source/op/custom_op.cc:17
         [[{node ProdEnvMatA}]]
         [[o_force/_29]]
  (1) INTERNAL: Operation received an exception: DeePMD-kit Error: CUDA Assert, in file /home/tools/LAMMPS-DeePMD-OCTP/deepmd-kit-v2.1.4/source/op/custom_op.cc:17
         [[{node ProdEnvMatA}]]
0 successful operations.
0 derived errors ignored. (../pair_deepmd.cpp:390)
Last command: run                       ${step_total}   upto

**Comment by goodluck-xyz** [comment]
I meet the same error. Have you solved this error?

**Reply by njzjz** [reply]
Are there any other messages from stderr? Currently I only understand the error comes from CUDA.

**Reply by goodluck-xyz** [reply]
my tensorflow is 2.9.0 and my suda is 11.7
the error log is presented below
Setting up Verlet run ...
  Unit style    : metal
  Current step  : 0
  Time step     : 0.001
INTERNAL: 2 root error(s) found.
  (0) INTERNAL: Blas xGEMV launch failed : a.shape=[1,3284736,25], b.shape=[1,1,25], m=3284736, n=1, k=25
         [[{{node gradients/filter_type_0/MatMul_grad/MatMul}}]]
         [[o_atom_virial/_25]]
  (1) INTERNAL: Blas xGEMV launch failed : a.shape=[1,3284736,25], b.shape=[1,1,25], m=3284736, n=1, k=25
         [[{{node gradients/filter_type_0/MatMul_grad/MatMul}}]]
0 successful operations.
0 derived errors ignored.
ERROR: DeePMD-kit Error: TensorFlow Error: INTERNAL: 2 root error(s) found.
  (0) INTERNAL: Blas xGEMV launch failed : a.shape=[1,3284736,25], b.shape=[1,1,25], m=3284736, n=1, k=25
         [[{node gradients/filter_type_0/MatMul_grad/MatMul}]]
         [[o_atom_virial/_25]]
  (1) INTERNAL: Blas xGEMV launch failed : a.shape=[1,3284736,25], b.shape=[1,1,25], m=3284736, n=1, k=25
         [[{node gradients/filter_type_0/MatMul_grad/MatMul}]]
0 successful operations.
0 derived errors ignored. (../pair_deepmd.cpp:389)
Last command: run             2000

**Reply by njzjz** [reply]
This is like #1062, but it should not happen in CUDA 11.7. You can check if any old libraries are used.

**Reply by goodluck-xyz** [reply]
I haven't figure out what's wrong. After I load the deepmd module, the lammps run successfully. However, I ran another lmp progroam successfully without load deepmd module. If I compute msd and rdf , is loading deepmd module necessary?

**Reply by njzjz** [reply]
Usually TensorFlow will log the version and the location of CUDA libraries. Could you find it?

**Reply by goodluck-xyz** [reply]
[error.log](https://github.com/deepmodeling/deepmd-kit/files/9886161/error.log)
you mean this?

**Reply by wxm0703** [reply]
I meet the same error too. Have you solved this error?

**Reply by goodluck-xyz** [reply]
Maybe `mpiru -np 4 lmp_mpi -i input.lammps` will help you.

**Comment by ChunyiZh** [comment]
You may also check to see whether this error is caused by the DPLR model or by the software by conducting the DPLR example calculation here: https://github.com/CSIprinceton/workshop-july-2022/tree/main/hands-on-sessions/day-2/5-deep-potential-long-range/3-md
It includes the DPLR model and the input script. If this simulation can be run correctly on your machine, there may be some problems related to your model or script.
